subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftBasePlugin.java

```diff
@@ -76,5 +90,25 @@ public void apply(ProjectInternal project) {
         nativeRuntime.extendsFrom(implementation);
         nativeRuntime.setCanBeConsumed(false);
         nativeRuntime.getAttributes().attribute(Usage.USAGE_ATTRIBUTE, project.getObjects().named(Usage.class, Usage.NATIVE_RUNTIME));
+
+        Configuration testImplementation = project.getConfigurations().create(TEST_IMPLEMENTATION);
+        testImplementation.extendsFrom(implementation);
```

Comment:
I added a TODO. I will address the convention with `swift-library` and `swift-executable` in a later PR.